core/trino-parser/src/main/java/io/trino/sql/tree/ExecuteImmediate.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class ExecuteImmediate
+        extends Statement
+{
+    private final StringLiteral statement;
+    private final List<Expression> parameters;
+
+    public ExecuteImmediate(NodeLocation location, StringLiteral statement, List<Expression> parameters)
+    {
+        super(Optional.of(requireNonNull(location, "location is null")));
+        this.statement = requireNonNull(statement, "statement is null");
+        this.parameters = ImmutableList.copyOf(requireNonNull(parameters, "parameters is null"));
```

Comment:
Same here. `ImmutableList.copyOf` will fail with NPE if parameters is null, so the check is unnecessary.